app/src/main/java/com/doctor/yumyum/presentation/ui/login/LoginViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.doctor.yumyum.presentation.ui.login
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.doctor.yumyum.common.base.BaseViewModel
+import com.doctor.yumyum.data.model.signUpModel
+import com.doctor.yumyum.data.repository.RepositoryImpl
+import com.google.gson.Gson
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.Response
+import retrofit2.http.Header
+
+
+class LoginViewModel : BaseViewModel() {
+    val repository: RepositoryImpl = RepositoryImpl()
+    private val _accessToken: MutableLiveData<String> = MutableLiveData()
+    val accessToken: LiveData<String>
+        get() = _accessToken
+    private val _nickname: MutableLiveData<String> = MutableLiveData()
+    val nickname: LiveData<String>
+        get() = _nickname
+    private val _oauthType: MutableLiveData<String> = MutableLiveData()
+    val oauthType: LiveData<String>
+        get() = _oauthType
+    private val _errorState: MutableLiveData<Boolean> = MutableLiveData(false)
+    val errorState: LiveData<Boolean>
+        get() = _errorState
+
+    fun setAccessToken(t: String) {
+        _accessToken.postValue(t)
+    }
+
+    fun setNickname(n: String) {
+        _nickname.postValue(n)
+    }
+
+    fun setOauthType(l: String) {
+        _oauthType.postValue(l)
+    }
+
+    fun signUp() {
+        repository.postAuthCreation(
+            signUpModel(
+                accessToken.value.toString(),
+                nickname.value.toString(),
+                oauthType.value.toString()
+            ),
```

Comment:
```suggestion
    fun signUp(accessToken: String, nickname: String, oauthType: String) {
        repository.postAuthCreation(
            SignUpModel(
                accessToken),
                nickname,
                oauthType
            ),
```

1. withContext 사용하면, 위처럼 변경할 수는 없을까요?
2. 1번 방향이 안된다고 할지라도 accesstoken, nickname, oauthType은 livedata 로 활용하지 않고 있어서, string type으로 선언해주는게 좋을 것 같아요 🙂